xapian-core/tests/harness/testsuite.cc

```diff
@@ -706,9 +711,20 @@ test_driver::do_run_tests(vector<string>::const_iterator b,
 		case PASS:
 		    ++res.succeeded;
 		    if (verbose || !use_cr) {
-			out << col_green << " ok" << col_reset << endl;
+			if (test_duration.count() >= 10000) {
+			    out << col_green << " ok" << col_yellow <<
+				" SLOW TEST (" << test_duration.count() <<
+				" ms)" << col_reset << endl;
+			} else {
+			    out << col_green << " ok" << col_reset << endl;
+			}
 		    } else {
-			out << "\r                                                                               \r";
+			string fill = "\r";
+			for (int i = 0; i < 90; ++i) {
+			    fill += " ";
+			}
+			fill += "\r";
+			out << fill;
```

Comment:
This change would have been better in its own commit, but actually I think it's a change we don't want in this form — it looks like you're dynamically constructing a `std::string`` one character at a time, which is almost certainly slower than the character constant previously. (You might be able to do better under C++11 using the `string(size_t n, char c)` constructor and then changing the first and last characters.)

Reply:
I agree with James that the current approach seems better.

The current code has 79 spaces - is there a reason to increase that to 90?  On a standard 80 column terminal, 90 will cause the terminal to scroll which will look odd.

Reply:
removed the change.